code/go/0chain.net/smartcontract/storagesc/allocation_write_pools.go

```diff
@@ -0,0 +1,67 @@
+package storagesc
+
+import (
+	chainstate "0chain.net/chaincore/chain/state"
+	"0chain.net/chaincore/state"
+	"0chain.net/core/common"
+	"errors"
+	"fmt"
+)
+
+// Created using StorageAllocation.getAllocationPools
+type allocationWritePools struct {
+	// The indices for ids and writePools match
+	ownerId         int
+	ids             []string
+	writePools      []*writePool
+	allocationPools allocationPools
+}
+
+func (awp *allocationWritePools) getOwnerWP() (*writePool, error) {
+	if len(awp.writePools) == 0 {
```

Comment:
It might have `nil pointer dereference` crash too.  do you think it is better to add 
```
if awp == nil {
return nil, errors.new("no write pool instance")
}
```

Reply:
This is always called after `awp, err := getAllocationPools` has not returned an error. This can never be nil in production. If it is ever nil there is a coding error that `must` be fixed.